forge/routes/api/device.js

```diff
@@ -70,13 +70,37 @@ module.exports = async function (app) {
     app.post('/', {
         preHandler: [
             async (request, reply) => {
-                // Attach the team membership to the request so needsPermission can
-                // verify the user has the required role
-                if (request.body && request.body.team) {
+                // * If this is a Device Provisioning action: verify the device has the required scope
+                //   & session has been populated with provisioning data
+                // * If this is a User action: verify the user has the required role
+                if (request.session.provisioning) {
+                    // A device is auto-provisioning. First check the request body team matches the token
+                    // NOTE: If the token was not valid, the request would have been rejected by
+                    // the verifySession decorator & request.session.provisioning would not be populated
+                    const teamOK = request.body.team && request.body.team === request.session.provisioning.team
+                    if (teamOK) {
+                        request.deviceProvisioning = true
```

Comment:
Is this strictly needed if `request.session.provisioning` can also be used to know this is a valid provisioning token?

Reply:
Maybe a bit of belt + braces but ultimately a comfort blanket 😄 
Removed the explicit `request.deviceProvisioning` flag in [8ff3f7d](https://github.com/flowforge/flowforge/pull/1659/commits/8ff3f7df313252912df229bb10440425e612c70e)